ui/components/app/transaction-list-item/transaction-list-item.component.js

```diff
@@ -242,7 +270,46 @@ export default function TransactionListItem({
   );
 }
 
-TransactionListItem.propTypes = {
+TransactionListItemInner.propTypes = {
   transactionGroup: PropTypes.object.isRequired,
   isEarliestNonce: PropTypes.bool,
+  setEditGasMode: PropTypes.func,
+};
+
+const TransactionListItem = (props) => {
+  const { transactionGroup } = props;
+  const [editGasMode, setEditGasMode] = useState();
+  const transaction = transactionGroup.primaryTransaction;
+  const EIP_1559_V2_ENABLED =
```

Comment:
Probably too late for this, but in the future we should probably make some kind of FeatureFlagContext that checks for feature flags and makes them available to all components rather than repeat this logic.

Reply:
In case of `EIP_1559_V2` condition is bit more complex as it requires checking that flag is enabled and also that network and account support EIP-1559 V2.